### ISSUE TITLE
Compile problem on MacOS: Undefined symbols for architecture arm64

### ISSUE THREAD

**Issue by adamlei** [vinecopulib/vinecopulib#551]
I am playing with vinecopula and try to implement the code in the website.

My machine with clang version is
Apple clang version 13.1.6 (clang-1316.0.21.2.3)
Target: arm64-apple-darwin21.4.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin

I followed the instruction from the website to install the vinecopulib successfully and unit test went no problem as well. 
However I tried to compile it with command: `clang++ -std=c++11 myvineCopula.cpp`
But it failed with error msg:

```
“Undefined symbols for architecture arm64:
  "vinecopulib::get_family_name(vinecopulib::BicopFamily)", referenced from:
      _main in myvineCopula-172b5e.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)”

```

Thank you so much for helping ! Please let me know if you need more information.

Here is my code: 
myvineCopula.cpp
```
#include` "myvineCopula.hpp"


int main(){
    std::cout << "Available families : ";
    for (auto family : vinecopulib::bicop_families::all) {
            std::cout << get_family_name(family) << " ";
    }
    return 0;
}

```
myvineCopula.hpp:

```
#ifndef myvineCopula_hpp
#define myvineCopula_hpp

#include <stdio.h>
#include "vinecopulib.hpp"
#include <iostream>

#endif /* myvineCopula_hpp */


```

**Comment by tnagler** [comment]
The function is defined [here](https://github.com/vinecopulib/vinecopulib/blob/master/include/vinecopulib/bicop/family.hpp). Did you try to compile the code in [`/examples`](https://github.com/vinecopulib/vinecopulib/tree/master/examples)?

**Comment by tvatter** [comment]
Did you install vinecopulib and made sure that the needed headers are in the clang++ path?

After the vinecopulib install, we generally use cmake to simplify the build process as shown in the examples folder. But a program can also be compiled manually as in your example, provided that the headers are in the path, e.g. with the -I argument.

**Comment by adamlei** [comment]
> The function is defined [here](https://github.com/vinecopulib/vinecopulib/blob/master/include/vinecopulib/bicop/family.hpp). Did you try to compile the code in [`/examples`](https://github.com/vinecopulib/vinecopulib/tree/master/examples)?


Hi. Yes I tried.  I went to the `/examples` dictionary and compile the `main.cpp` and get the error below:

```
Undefined symbols for architecture arm64:
  "vinecopulib::get_family_name(vinecopulib::BicopFamily)", referenced from:
      _main in main-0d2d93.o
  "vinecopulib::FitControlsBicop::FitControlsBicop(std::__1::vector<vinecopulib::BicopFamily, std::__1::allocator<vinecopulib::BicopFamily> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, double, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, Eigen::Matrix<double, -1, 1, 0, -1, 1> const&, double, bool, unsigned long)", referenced from:
      _main in main-0d2d93.o
  "vinecopulib::Bicop::Bicop(vinecopulib::BicopFamily, int, Eigen::Matrix<double, -1, -1, 0, -1, -1> const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&)", referenced from:
      _main in main-0d2d93.o
  "vinecopulib::Bicop::Bicop(Eigen::Matrix<double, -1, -1, 0, -1, -1> const&, vinecopulib::FitControlsBicop const&, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > const&)", referenced from:
      _main in main-0d2d93.o
  "vinecopulib::FitControlsBicop::str() const", referenced from:
      _main in main-0d2d93.o
  "vinecopulib::Bicop::str() const", referenced from:
      _main in main-0d2d93.o
  "vinecopulib::Bicop::simulate(unsigned long const&, bool, std::__1::vector<int, std::__1::allocator<int> > const&) const", referenced from:
      _main in main-0d2d93.o
ld: symbol(s) not found for architecture arm64
```

This msg gives me no clue about how to fix it. Thank you so much in advance for being patience in helping.

**Comment by adamlei** [comment]
> Yes, clang++ doesn't know where to find the library. See tvatter's comment above.
> 
> Especially if you're new to C++ I would advise to start from the cmake projects in the `/examples` folder. For example, from `examples/bicop/` run `cmake . && make && ./bin/main` to compile and run the example.

tanglier and tvatter, Thank you so much for suggestions!  `cmake . && make && ./bin/main` make it work!! I think tvatter was right about  _**"needed headers are in the clang++ path"**_ 

Therefore, I try to track down the clang include path.  run `clang++ -std=c++11 main.cpp -v ` and get 
```
#include "..." search starts here:
#include <...> search starts here:
 /usr/local/include
 /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1

```

However, the vinecopula.hpp is found in this path ` /usr/local/include`

```
Eigen			itclDecls.h		tclOO.h			tkDecls.h
boost			itclInt.h		tclOODecls.h		tkPlatDecls.h
eigen3			itclIntDecls.h		tclPlatDecls.h		unsupported
fakemysql.h		itclMigrate2TclCore.h	tclThread.h		vinecopulib
fakepq.h		itclTclIntStubsFcn.h	tclTomMath.h		vinecopulib.hpp
fakesql.h		mysqlStubs.h		tclTomMathDecls.h	wdm
gmock			odbcStubs.h		tdbc.h			wdm.hpp
gtest			pqStubs.h		tdbcDecls.h
itcl.h			tcl.h			tdbcInt.h
itcl2TclOO.h		tclDecls.h		tk.h
```

Now I am more confused why clang＋＋ cannot find it?

Thank you so much for the helping.

**Comment by tnagler** [comment]
That explains why the library compiles. The fact that clang complains about missing "symbols" indicates that it fails during linking. It looks for compiled versions of the functions and can't find it. Did you compile with `-DVINECOPULIB_SHARED_LIB=ON` before installing? In that case, you also need to tell clang where the binaries are.

**Comment by adamlei** [comment]
> That explains why the library compiles. The fact that clang complains about missing "symbols" indicates that it fails during linking. It looks for compiled versions of the functions and can't find it. Did you compile with `-DVINECOPULIB_SHARED_LIB=ON` before installing? In that case, you also need to tell clang where the binaries are.

Thanks for explaining ! I did use `cmake ..` instead of `cmake .. -DVINECOPULIB_SHARED_LIB=ON` since I was not really sure about ` -DVINECOPULIB_SHARED_LIB=ON` !!
Should I reinstall the vinecopulib ? I did not find anything relevant for letting Clang find the complied versions.

Thank you again for the help !

**Comment by tnagler** [comment]
Again, I just recommend using cmake, if you don't know how to troubleshoot a custom clang compilation.

**Comment by adamlei** [comment]
> Did you install vinecopulib and made sure that the needed headers are in the clang++ path?
> 
> After the vinecopulib install, we generally use cmake to simplify the build process as shown in the examples folder. But a program can also be compiled manually as in your example, provided that the headers are in the path, e.g. with the -I argument.

Hi tvatter, after these days, I tried to use the vinecopulib in a R (but functions are mainly written in C) project.  But the problem still exists so I came back to the original point that clang cannot find the location of the library.

I reinstalled the package with `cmake .. -DVINECOPULIB_SHARED_LIB=ON` as suggested by tnagler and run the code with `-I` argument with the location of the library but it still failed.

What I tried are

`clang++ myvineCopula.cpp -I /usr/local/include/vinecopulib/bicop/family.hpp`
` clang++ -std=c++11 myvineCopula.cpp -L /usr/local/lib`
`clang++ -std=c++11 myvineCopula.cpp -I /usr/local/include`

Where I did is wrong? Why I cannot compiled it manually?
Thank you so much for your help!
